### PR TITLE
Changed white-space handling in text elements

### LINKED ISSUE #10542: Justify Text Alignment Off in FireFox Browser
## Bug Description
Justify text behavior seems off/different in Firefox browser versus Chrome browser. View [this example story](https://testsite.cmsdevrel.com/web-stories/justified-text-test/) in Chrome then compare it in Firefox.

## Expected Behaviour

For the justify text alignment to apply accordingly. 

## Steps to Reproduce

1. Open Chrome browser 
2. Add Text > Justify text
2. View it in Chrome. Then view it in Firefox.
3. Open Firefox browser
4. Add Text > Justify text
5. Note behavior

## Screenshots
Firefox:
Screen Shot 2022-02-10 at 4 19 20 PM
Chrome:
Screen Shot 2022-02-10 at 4 19 53 PM

## Additional Context
related topic: https://wordpress.org/support/topic/justified-button-not-working-in-editor/

- Plugin Version: 1.17.0
- WordPress Version:
- Operating System: macOS
- Browser: Firefox

### PR DESCRIPTION
**Pull request by barklund**
## Summary

This changes how white-space is handled in text elements, as Firefox is unable to make justified text, if white-space is set tp `pre-wrap`. Setting it to `pre-line` does (almost) the same thing while avoiding the bug in Firefox completely.

The main difference between `pre-wrap` and `pre-line` are:

- Multiple white-space characters in a row are collapsed (unless non-breaking).
- Leading white-space is removed (unless non-breaking).

Potential consequences:

- If any text sets use either of the above, they would render wrong. However, both searching as well as manual checking seem to indicate, that no text sets do this.
- If any templates (story and/or page) use either of the above, they would render wrong. It was a lot harder to check these, but it does not appear that any of them use either of the above.
- If a user has a story with multiple spaces or leading space in a line, it could potentially render wrong. However, because our text editor converts all spaces to non-breaking spaces and all tabs (they could be pasted in) to four non-breaking spaces, this is never actually a problem. Multiple spaces manually entered in the editor still render correctly!

## User-facing changes

Various browsers with various text alignment, before and after:

| Browser | Before | After |
|-|-|-|
| Chrome | chrome_before | chrome_after |
| Safari | safari_before | safari_after |
| Firefox | firefox_before | firefox_after |

Note however how different browsers treat multiple spaces differently! Between the red letters, there are two (non-breaking) spaces, between the other words only one space. The differences are not completely logical to me, but definitely there:
pre-line-rendering-differences

## Testing Instructions

This PR can be tested by following these steps:

1.  Pay special attention to Firefox, but test in other browsers too
2. Add a text field and make sure it has multiple lines due to wrapping, not due to manual line-breaks
3. Adjust text alignment to justified
4. Observe that text actually renders justified in both editor and final story
5. Play with multiple spaces, newlines, and leading spaces after a newline and observe, that all white-space renders correctly in both editor and final story

Known bug with Firefox + Justified Text + Edit-mode:

1. Add a text field in Firefox
2. Add multiple lines of text
3. Set text alignment to justified
4. Click inside text to enter edit mode
5. Observe that text is not longer justified but left-aligned. This is a known and acceptable bug and it is better than the alternative.
6. Exit editmode and observe that text once again appears justified
7. Preview the story
8. Observe that text is correctly justified

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #10542

### CODE REVIEW

**Review by merapi — COMMENTED** [review]
I'm still checking the impact of this.

Is it just as easy to add this to the rich text editor? To make the display and edit mode look the same.

https://user-images.githubusercontent.com/21036224/163474814-12669509-d31c-4aaa-b4ff-38613e26a608.mov

**Comment by barklund** [comment]
> I'm still checking the impact of this.

No worries, it's a tiny change with a potential for a lot of consequences! So let's be diligent before merging!

> Is it just as easy to add this to the rich text editor? To make the display and edit mode look the same.

Excellent point, forgot about that!

File: packages/element-library/src/text/edit.js [hunk 1]
```diff
@@ -67,6 +67,16 @@ import {
...
+// white-space: pre-wrap is hardcoded with an inline style inside draft-js,
+// but we can override it with a global important style on the proper element.
+// @see https://github.com/GoogleForCreators/web-stories-wp/pull/11262
+// @see https://github.com/facebook/draft-js/blob/d974abb0c2844bf99724aba9448df42a63291e0e/src/component/base/DraftEditor.react.js#L354
+const GlobalStyleOverride = createGlobalStyle`
+  .public-DraftEditor-content {
+    white-space: pre-line !important;
+  }
+`;
+
```
Comment (barklund) [inline packages/element-library/src/text/edit.js]: @merapi, it wasn't the prettiest fix, but it works consistently.

**Review by BrittanyIRL — DISMISSED** [review]
I think this looks good! What an interesting browser difference.

**Comment by merapi** [comment]
> What an interesting browser difference.

Yeah, browsers are really interesting piece of software 😉 

image

**Review by merapi — CHANGES_REQUESTED** [review]
This is waiting for [UX/product decision](https://xwp.slack.com/archives/CCCAUAH6F/p1650483814965059?thread_ts=1650453276.486579&cid=CCCAUAH6F) due to this bug (removing whitespaces):


https://user-images.githubusercontent.com/21036224/164794608-4d902e85-fbd0-443f-bb14-3cb73a3ea242.mp4

**Review by merapi — DISMISSED** [review]

**Review by merapi — APPROVED** [review]

**Pull request merged**